src/overlays/actors/ovl_En_Bat/z_en_bat.h

```diff
@@ -3,14 +3,41 @@
 
 #include "global.h"
 
+#define BAD_BAT_GET_TYPE(thisx) (((thisx)->params >> 0) & 0x1F)
+#define BAD_BAT_GET_PARAMFLAGS(thisx) (((thisx)->params >> 5) & 7)
+#define BAD_BAT_GET_SWITCHFLAG(thisx) (((thisx)->params >> 8) & 0xFF)
+#define BAD_BAT_PARAMS(switchFlag, paramflags, type) ((type) | ((paramflags) << 5) | ((switchFlag) << 8))
+
+//! PARAMFLAGS
+//! Affects which flags are used in Actor_UpdateBgCheckInfo in certain circumstances
+#define BAD_BAT_PARAMFLAG_0 (1 << 0)
+//! Whether to apply a height range check to attack
+#define BAD_BAT_PARAMFLAG_CHECK_HEIGHTREL (1 << 1)
+//! Spawn as perching instead of flying
+#define BAD_BAT_PARAMFLAG_PERCH (1 << 2)
+
+#define BAD_BAT_GET_NUMBER_TO_SPAWN(thisx) ((thisx)->params)
```

Comment:
I though we don't make macros for just straight params access?

Reply:
In this case it strips all the other stuff off first, then does arithmetic on it. It is a reasonable question what to do with the other parts.